2025-03-20 15:52:49,743 ERROR: Exception on /index [GET] [in C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\app.py:1454]
Traceback (most recent call last):
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\neggm\DBWE\app\routes.py", line 34, in index
    return render_template('index.html', title='Home', form=form, posts=posts.items, next_url=next_url, prev_url=prev_url)
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\templating.py", line 147, in render_template
    return _render(
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\jinja2\environment.py", line 1295, in render
    self.environment.handle_exception()
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\jinja2\environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\neggm\DBWE\app\templates\index.html", line 2, in top-level template code
    {% extends "base.html" %}
  File "C:\Users\neggm\DBWE\app\templates\base.html", line 2, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 1, in top-level template code
    {% block doc -%}
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 4, in block 'doc'
    {%- block html %}
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in block 'html'
    {% block body -%}
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 23, in block 'body'
    {% block content -%}
  File "C:\Users\neggm\DBWE\app\templates\base.html", line 47, in block 'content'
  File "C:\Users\neggm\DBWE\app\templates\index.html", line 6, in block 'app_content'
    {{ wtf.quick_form(form) }}
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\jinja2\environment.py", line 490, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'wtf' is undefined
2025-03-20 15:53:05,051 ERROR: Exception on /index [GET] [in C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\app.py:1454]
Traceback (most recent call last):
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\app.py", line 2070, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\app.py", line 1515, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\app.py", line 1513, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\app.py", line 1499, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\neggm\DBWE\app\routes.py", line 34, in index
    return render_template('index.html', title='Home', form=form, posts=posts.items, next_url=next_url, prev_url=prev_url)
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\templating.py", line 147, in render_template
    return _render(
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\jinja2\environment.py", line 1295, in render
    self.environment.handle_exception()
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\jinja2\environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\neggm\DBWE\app\templates\index.html", line 2, in top-level template code
    {% extends "base.html" %}
  File "C:\Users\neggm\DBWE\app\templates\base.html", line 2, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 1, in top-level template code
    {% block doc -%}
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 4, in block 'doc'
    {%- block html %}
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in block 'html'
    {% block body -%}
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 23, in block 'body'
    {% block content -%}
  File "C:\Users\neggm\DBWE\app\templates\base.html", line 47, in block 'content'
  File "C:\Users\neggm\DBWE\app\templates\index.html", line 6, in block 'app_content'
    {{ wtf.quick_form(form) }}
  File "C:\Users\neggm\DBWE\myenv\lib\site-packages\jinja2\environment.py", line 490, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'wtf' is undefined
2025-03-20 15:55:27,195 INFO: Microblog startup [in C:\Users\neggm\DBWE\app\__init__.py:29]
2025-03-20 15:57:47,976 INFO: Microblog startup [in C:\Users\neggm\DBWE\app\__init__.py:29]
2025-03-20 15:59:08,445 INFO: Microblog startup [in C:\Users\neggm\DBWE\app\__init__.py:29]
2025-03-20 15:59:58,443 INFO: Microblog startup [in C:\Users\neggm\DBWE\app\__init__.py:29]
2025-03-20 16:32:39,149 INFO: Microblog startup [in /home/student/DBWE/app/__init__.py:29]
2025-03-20 16:38:21,009 INFO: Microblog startup [in /home/student/DBWE/app/__init__.py:29]
2025-03-20 16:38:21,221 INFO: Microblog startup [in /home/student/DBWE/app/__init__.py:29]
2025-03-20 16:38:21,323 INFO: Microblog startup [in /home/student/DBWE/app/__init__.py:29]
2025-03-20 16:38:21,392 INFO: Microblog startup [in /home/student/DBWE/app/__init__.py:29]
2025-03-21 00:21:33,445 INFO: Microblog startup [in /home/student/DBWE/app/__init__.py:29]
2025-03-21 00:48:58,914 INFO: Microblog startup [in /home/student/DBWE/app/__init__.py:29]
2025-03-21 00:48:58,928 INFO: Microblog startup [in /home/student/DBWE/app/__init__.py:29]
2025-03-21 00:48:59,091 INFO: Microblog startup [in /home/student/DBWE/app/__init__.py:29]
2025-03-21 00:48:59,141 INFO: Microblog startup [in /home/student/DBWE/app/__init__.py:29]
2025-03-21 00:53:02,006 INFO: Microblog startup [in /home/student/DBWE/app/__init__.py:29]
